backend/src/main/java/org/example/restaurantmanagementapplication/rest/OrderController.java

```diff
@@ -1,65 +1,105 @@
 
 @RestController
 @RequestMapping("/orders")
 @RequiredArgsConstructor
 public class OrderController {
 
   private final OrderService orderService;
+  private final OrderMapper mapper;
   private final SessionManager sessionManager;
 
   @GetMapping
-  public ResponseEntity<List<Order>> retrieveAllOrders() {
-    return ResponseEntity.ok(orderService.findAll());
+  public ResponseEntity<List<OrderDto>> retrieveAllOrders() {
+    return ResponseEntity.ok(orderService.findAll().stream().map(mapper::toDTO).toList());
   }
 
   @PostMapping
-  public ResponseEntity<Order> createOrder(@RequestBody List<OrderItemRequest> items) {
-    Optional<User> currentUser = sessionManager.getCurrentUser();
-
-    var createdBy = currentUser.isPresent()
-        ? currentUser.get().getEmail()
-        : "anonymous";
-
+  public ResponseEntity<OrderDto> createOrder(@RequestBody List<OrderItemRequest> items) {
     var orderRequest = new OrderRequest();
     orderRequest.setOrderedItems(items);
-    orderRequest.setCreatedBy(createdBy);
-    Order order = orderService.createOrder(orderRequest);
-    return ResponseEntity.ok(order);
+    orderRequest.setCreatedBy(getCurrentUserName());
+    var order = orderService.createOrder(orderRequest);
+    return ResponseEntity.ok(mapper.toDTO(order));
   }
 
   @PostMapping("{id}/cancel")
-  public ResponseEntity<Order> cancelOrder(@PathVariable int id) {
-    Optional<User> currentUser = sessionManager.getCurrentUser();
-
-    var updatedBy = currentUser.isPresent()
-        ? currentUser.get().getEmail()
-        : "anonymous";
-
-    Order order = orderService.findById(id);
+  public ResponseEntity<OrderDto> cancelOrder(@PathVariable int id) {
+    var order = orderService.findById(id);
     if (order == null) {
       return ResponseEntity.notFound().build();
     }
     if (!OrderStatusTransition.isCancellable(order.getStatus())) {
-      return ResponseEntity.badRequest().body(order);
+      return ResponseEntity.badRequest().body(mapper.toDTO(order));
     }
     order.setStatus(OrderStatus.CANCELLED);
-    order.setUpdatedBy(updatedBy);
+    order.setUpdatedBy(getCurrentUserName());
+    order.setUpdatedAt(Instant.now());
     orderService.save(order);
-    return ResponseEntity.ok(order);
+    return ResponseEntity.ok(mapper.toDTO(order));
+  }
+
+
+  @PostMapping("{id}/pending")
+  public ResponseEntity<OrderDto> setPending(@PathVariable int id) {
+    return updateStatus(id, OrderStatus.PENDING);
+  }
+
+  @PostMapping("{id}/confirmed")
+  public ResponseEntity<OrderDto> setConfirmed(@PathVariable int id) {
+    return updateStatus(id, OrderStatus.CONFIRMED);
+  }
+
+  @PostMapping("{id}/in_progress")
+  public ResponseEntity<OrderDto> setInProgress(@PathVariable int id) {
+    return updateStatus(id, OrderStatus.IN_PROGRESS);
+  }
+
+  @PostMapping("{id}/ready")
+  public ResponseEntity<OrderDto> setReady(@PathVariable int id) {
+    return updateStatus(id, OrderStatus.READY);
+  }
```

Comment:
A general approach regarding API endpoints for state transitions is to use action names. That way, we do not expose internal logic to the client. I would propose:
```suggestion
  @PostMapping("{id}/mark-as-pending")
  public ResponseEntity<OrderDto> setPending(@PathVariable int id) {
    return updateStatus(id, OrderStatus.PENDING);
  }

  @PostMapping("{id}/confirm")
  public ResponseEntity<OrderDto> setConfirmed(@PathVariable int id) {
    return updateStatus(id, OrderStatus.CONFIRMED);
  }

  @PostMapping("{id}/start-preparation")
  public ResponseEntity<OrderDto> setInProgress(@PathVariable int id) {
    return updateStatus(id, OrderStatus.IN_PROGRESS);
  }

  @PostMapping("{id}/mark-as-ready")
  public ResponseEntity<OrderDto> setReady(@PathVariable int id) {
    return updateStatus(id, OrderStatus.READY);
  }
```
or something similar in the case of `setInProgress()` method.
Aside from that, all good! I like the parametrized tests :)

Reply:
Applied